rpm/src/main/java/org/eclipse/packager/rpm/signature/RpmFileSignatureProcessor.java

```diff
@@ -0,0 +1,208 @@
+            processor.feedHeader(headerBuf.slice());
+            processor.feedPayloadData(payloadBuf.slice());
+            processor.finish(signatureHeader);
+        }
+        ByteBuffer signatureBuf = Headers.render(signatureHeader.makeEntries(), true, Rpms.IMMUTABLE_TAG_SIGNATURE);
+        final int payloadSize = signatureBuf.remaining();
+        final int padding = Rpms.padding(payloadSize);
+        byte[] signature = safeReadBuffer(signatureBuf);
+        ByteArrayOutputStream result = new ByteArrayOutputStream();
+        result.write(signature);
+        if (padding > 0) {
+            result.write(safeReadBuffer(ByteBuffer.wrap(Rpms.EMPTY_128, 0, padding)));
+        }
+        return result.toByteArray();
+    }
+
+    /**
+     * <p>
+     * Safe read (without buffer bytes) the given buffer and return it to a byte
+     * array
+     * </p>
+     * 
+     * @param buf : the {@link ByteBuffer} to read
+     * @return byte array
+     */
+    private byte[] safeReadBuffer(ByteBuffer buf) {
+        ByteArrayOutputStream result = new ByteArrayOutputStream();
+        while (buf.hasRemaining()) {
+            result.write(buf.get());
+        }
+        return result.toByteArray();
+    }
+
+    /**
+     * <p>
+     * Return all default {@link SignatureProcessor} defined in
+     * {@link SignatureProcessors}
+     * </p>
+     * 
+     * @return {@link List<SignatureProcessor>} of {@link SignatureProcessor}
+     */
+    private List<SignatureProcessor> getDefaultsSignatureProcessors() {
+        List<SignatureProcessor> signatureProcessors = new ArrayList<>();
+        signatureProcessors.add(SignatureProcessors.size());
+        signatureProcessors.add(SignatureProcessors.sha256Header());
+        signatureProcessors.add(SignatureProcessors.sha1Header());
+        signatureProcessors.add(SignatureProcessors.md5());
+        signatureProcessors.add(SignatureProcessors.payloadSize());
+
+        return signatureProcessors;
+    }
+
+    /**
+     * <p>
+     * Convert an array of bytes into a ByteBuffer
+     * </p>
+     * 
+     * @param data : byte array to convert
+     * @return a {@link ByteBuffer} built with data
+     * @throws IOException
+     */
+    private ByteBuffer bufBytes(byte[] data) throws IOException {
+        ByteBuffer buf = ByteBuffer.allocate(data.length);
+        ReadableByteChannel headerChannel = Channels.newChannel(new ByteArrayInputStream(data));
+        IOUtils.readFully(headerChannel, buf);
+        return buf;
+    }
+
+    /**
+     * <p>
+     * Parse the byte[] to an RpmInputStream
+     * </p>
+     * 
+     * @param buf : byte array representing the rpm file
+     * @return {@link RpmInputStream}
+     * @throws IOException
+     */
+    private RpmInputStream getRpmInputStream(byte[] buf) throws IOException {
+        RpmInputStream ref = new RpmInputStream(new ByteArrayInputStream(buf));
+        ref.available(); // init RpmInputStream
+        ref.close();
```

Comment:
What is going on here with this method? And this only seems to be used once, why not remove it and better to use

```java
try (RpmInputStream ref = new RpmInputStream(new ByteArrayInputStream(buf))) {

}
```

Seems like best practice to me even though it isn't strictly necessary for byte arrays.

Reply:
This method is here only for the lisibility of the perform method.

Reply:
I put the try-with-resource but I prefers to keep this method.